helix-term/src/commands/typed.rs

```diff
@@ -914,189 +870,163 @@ fn theme(
 
 fn yank_main_selection_to_clipboard(
     cx: &mut compositor::Context,
-    _args: &[Cow<str>],
+    _args: Args,
     event: PromptEvent,
 ) -> anyhow::Result<()> {
     if event != PromptEvent::Validate {
         return Ok(());
     }
-
     yank_primary_selection_impl(cx.editor, '+');
     Ok(())
 }
 
-fn yank_joined(
-    cx: &mut compositor::Context,
-    args: &[Cow<str>],
-    event: PromptEvent,
-) -> anyhow::Result<()> {
+fn yank_joined(cx: &mut compositor::Context, args: Args, event: PromptEvent) -> anyhow::Result<()> {
     if event != PromptEvent::Validate {
         return Ok(());
     }
-
-    ensure!(args.len() <= 1, ":yank-join takes at most 1 argument");
-
-    let doc = doc!(cx.editor);
-    let default_sep = Cow::Borrowed(doc.line_ending.as_str());
-    let separator = args.first().unwrap_or(&default_sep);
     let register = cx.editor.selected_register.unwrap_or('"');
-    yank_joined_impl(cx.editor, separator, register);
+    yank_joined_impl(cx.editor, args.raw(), register);
```

Comment:
hmm I think you need to look trough at this function again:

this function takes 0 or 1 arguments (last argument is optional) and in case there are zero arguments then we want to just insert the document line ending.

Reply:
Is this only for `yank_joined`? or all `yank_joined_impl`? If its all I can move this check to that.

Reply:
it's for all of them

Reply:
Yeah, looking at the code more closely now, I remembered I removed the `ensure!` as `raw` would always return something, even if its just an empty `&str`, so there is no check for multiple args as all "args" would be treated as a single, full string, making the check there pointless.